cwms-data-api/src/main/java/cwms/cda/data/dao/ForecastSpecDao.java

```diff
@@ -1,38 +1,124 @@
 
-public class ForecastSpecDao extends JooqDao<ForecastSpec> {
+public final class ForecastSpecDao extends JooqDao<ForecastSpec> {
 
     public ForecastSpecDao(DSLContext dsl) {
         super(dsl);
     }
 
 
     public void create(ForecastSpec forecastSpec) {
-        throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+        connection(dsl, conn -> {
+            setOffice(conn, forecastSpec.getOfficeId());
 
+            String timeSeriesIds = null;
+            if (forecastSpec.getTimeSeriesIds() != null) {
+                timeSeriesIds = String.join("\n", forecastSpec.getTimeSeriesIds());
+            }
+            CWMS_FCST_PACKAGE.call_STORE_FCST_SPEC(DSL.using(conn).configuration(), forecastSpec.getSpecId(),
+                    forecastSpec.getDesignator(), forecastSpec.getSourceEntityId(),
+                    forecastSpec.getDescription(), forecastSpec.getLocationId(),
+                    timeSeriesIds, "F", "F", forecastSpec.getOfficeId());
+        });
     }
 
-    public void delete(String office, String specId, String designator) {
-        throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+    public void delete(String office, String specId, String designator, DeleteRule deleteRule) {
+        connection(dsl, conn -> {
+            setOffice(conn, office);
+            CWMS_FCST_PACKAGE.call_DELETE_FCST_SPEC(DSL.using(conn).configuration(), specId, designator,
+                    deleteRule.getRule(), office);
+        });
     }
 
     public List<ForecastSpec> getForecastSpecs(String office, String specIdRegex,
-                                               String designator, String location,
-                                               String sourceEntity) {
-        throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+            String designator, String sourceEntity) {
+        AV_FCST_SPEC spec = AV_FCST_SPEC.AV_FCST_SPEC;
+        return forecastSpecQuery(dsl)
+                .where(JooqDao.caseInsensitiveLikeRegex(spec.OFFICE_ID, office))
+                .and(JooqDao.caseInsensitiveLikeRegex(spec.FCST_SPEC_ID, specIdRegex))
+                .and(JooqDao.caseInsensitiveLikeRegex(spec.FCST_DESIGNATOR, designator))
+                .and(JooqDao.caseInsensitiveLikeRegex(spec.ENTITY_ID, sourceEntity))
+                .fetch()
+                .map(ForecastSpecDao::map);
+    }
+
+    private static SelectOnConditionStep<Record7<String, String, String, String,
+            String, String, String>> forecastSpecQuery(DSLContext dsl) {
+        AV_FCST_SPEC spec = AV_FCST_SPEC.AV_FCST_SPEC;
+        AV_FCST_TIME_SERIES timeSeries = AV_FCST_TIME_SERIES.AV_FCST_TIME_SERIES;
+        AV_FCST_LOCATION loc = AV_FCST_LOCATION.AV_FCST_LOCATION;
+        //Group all the timeseries ids into a "\n" delimited list
+        Table<Record2<String, String>> tsidTable = dsl.select(timeSeries.FCST_SPEC_CODE,
+                        DSL.listAgg(timeSeries.CWMS_TS_ID, "\n")
```

Comment:
interesting. haven't seen this done before

Reply:
AI prompt gave me that suggestion. I saw it also has a usage in the cwms_util package